src/Makefile

```diff
@@ -15,9 +15,14 @@
 release_hdr := $(shell sh -c './mkreleasehdr.sh')
 uname_S := $(shell sh -c 'uname -s 2>/dev/null || echo not')
 uname_M := $(shell sh -c 'uname -m 2>/dev/null || echo not')
+CLANG := $(findstring clang,$(shell sh -c '$(CC) --version | head -1'))
 OPTIMIZATION?=-O3
 ifeq ($(OPTIMIZATION),-O3)
-	REDIS_CFLAGS+=-flto=auto
+	ifeq (clang,$(CLANG))
+		REDIS_CFLAGS+=-flto
+	else
+		REDIS_CFLAGS+=-flto=auto
+	endif
```

Comment:
If `-flto=auto` is GCC specific, perhaps we should instead use a check for GCC and use `-flto` for all other compilers. (Assuming there might be other compilers too.)

Reply:
@zuiderkwast Whether `-flto=auto` is GCC-specific (or whether it is accepted or not by other compilers) is unknown. All we know is Clang does not accept it, and we address that. If it turns out other compilers complain about it as well, we'll probably add compiler specific code.
I recommend something like:
```
# assume $(CCID) identifiers compiler: i.e., gcc, clang
CC_LTO_FLAGS.gcc=-flto=auto
CC_LTO_FLAGS.clang=-flto
CC_LTO_FLAGS=$(CC_LTO_FLAGS.$(CCID))
REDIS_CFLAGS += $(CC_LTO_FLAGS)
```

Reply:
i tend to agree with Viktor, if that `auto` thing is non-standard, let's just explicitly use it on GCC.
not sure it is necessary to build a big infrastructure around it, an if-else like you have now seems simpler.

Reply:
@oranagra IMO "standard" is not the right term here. We currently make assumptions on known behavior of Clang and everything that's not Clang (intelcc comes to mind, but it follows gcc frontend). Other exotic systems may have have completely other functions. So if we may try to detect gcc (which is tricky as I mentioned in a comment below) but this will over-complicate things (e.g. the intelcc case).
So I suggest the current solution, which is explicitly detecting clang, and betting other compilers to support `-flto=auto` until proved otherwise.

Reply:
`CC` can be `gcc`, `clang` or even some custom name (e.g. coming from a custom gcc build or Anaconda), so the best way to go is to invoke `$(CC) --version | head -1` and look for patterns (this can be made robust enough).
This will also solve the `gcc` invoking `clang` on macOS.

Reply:
my intention was that `-flto` is more "standard" that `-flto=<anything>`,
but ok, let's take your approach and see what the future brings.